var/spack/repos/builtin/packages/swak4foam/package.py

```diff
@@ -0,0 +1,203 @@
+# Please also see the LICENSE file for the LLNL notice and the LGPL.
+#
+# License
+# -------
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+#
+# Legal Notice
+# ------------
+# OPENFOAM is a trademark owned by OpenCFD Ltd
+# (producer and distributor of the OpenFOAM software via www.openfoam.com).
+# The trademark information must remain visible and unadulterated in this
+# file and via the "spack info" and comply with the term set by
+# http://openfoam.com/legal/trademark-policy.php
+#
+# This file is not part of OpenFOAM, nor does it constitute a component of an
+# OpenFOAM distribution.
+#
+##############################################################################
+from spack import *
+from spack.environment import *
+import llnl.util.tty as tty
+
+import multiprocessing
+import os
+from os.path import isfile
+
+
+#: Could move this out to openfoam-com utilities
+def find_foam_projectdir(foam_spec):
+    """Find the directory containing the OpenFOAM etc/bashrc file.
+    This directory corresponds to the WM_PROJECT_DIR.
+    Use a maxdepth of 1 in the search (for the directory).
+    Return None on failure.
+    """
+    testfile = join_path("etc", "bashrc")
+
+    if isfile(join_path(foam_spec.prefix, testfile)):
+        return foam_spec.prefix   # Trivial: located in prefix directly
+
+    # Check first level of sub-directories
+    with working_dir(foam_spec.prefix):
+        for d in os.listdir("."):
+            if not d.startswith('.') and isfile(join_path(d, testfile)):
+                return join_path(foam_spec.prefix, d)
+
+    raise RuntimeError('No OpenFOAM project directory found')
+
+
+class Swak4foam(Package):
+    """swak4Foam (SWiss Army Knife for Foam) supplies additional libraries
+    and utilities for OpenFOAM.
+    This offering is not approved or endorsed by OpenCFD Ltd,
+    producer and distributor of the OpenFOAM software via www.openfoam.com,
+    and owner of the OPENFOAM trademark.
+    """
+
+    homepage = "https://openfoamwiki.net/index.php/Contrib/swak4Foam"
+
+    version(
+        'dev',
+        branch='branches/develop',
+        git='https://github.com/Unofficial-Extend-Project-Mirror/openfoam-extend-swak4Foam-dev.git') # noqa
+    # The git mirror can be more convenient than using hg
+    # http://hg.code.sf.net/p/openfoam-extend/swak4Foam
```

Comment:
Yes, shallow clone and with git 'branch' I have a fair better idea of what I am getting. With hg, I'm also totally illiterate and can't make any upstream adjustments without creating complete chaos.